### ISSUE TITLE
Long-form audio transcription: model repeatedly outputs earlier segments (hallucination)

### ISSUE THREAD

**Issue by jackyguo624** [FireRedTeam/FireRedAudio#5]
Hi FireRedTeam,

First of all, FireRedAudio is a great piece of work, as were the previous models in the FireRed series. Thank you for open-sourcing the models and code — it benefits the whole community.

However, when I recently tried long-form audio transcription, the model repeated earlier parts of the transcription many times. This may be what is referred to as hallucination.

Here is my scritpt:
```bash
uv sync
python inference.py --task asr \
         --model /Pretrain_models/FireRedAudio \
         --audio  /data/aishell-4/sample_mono/M_R003S01C01_ch0.wav \
         --device "cuda:0" \
         --max-new-tokens "16384" \
```
For your reference, I have uploaded the mono AISHELL-4 sample wav to Google Drive: [M_R003S01C01_ch0.wav](https://drive.google.com/file/d/1z-IC-Dn3v0jfHfrYDrSmAyCEpfL9ShXs/view?usp=drive_link)
And got the transcription, 
[M_R003S01C01_ch0-000.hyp.txt](https://github.com/user-attachments/files/31905214/M_R003S01C01_ch0-000.hyp.txt)

You will find that starting from 12:34, the transcription repeats the 06:30–12:34 segment ("去小区宣传 → 年龄卡两周 → 晚接孩子/周六开园 → 保安发言/接送卡") until the end, and the output ends with a partial timestamp "[66".

I have also transcribed many other long-form audio files, and the repetition consistently appears:
[M_R003S02C01_ch0.wav](https://drive.google.com/file/d/1GObagYpan0a60b7V14uqXWTHpuXvey-R/view?usp=drive_link) which repeat “嗯” till the end

My questions are:
1. Is my script correct? Did I miss something or make a mistake?
2. As we know, FireRedASR-2s performs really well on Mandarin dialects and multilingual speech, so why is whisper-large-v3 used as the encoder here instead of the FireRedASR Conformer encoder? Is Whisper-style 30s chunked processing necessary for long-form transcription? (By the way [Moss-TD](https://github.com/OpenMOSS/MOSS-Transcribe-Diarize) uses the same encoder)
3. Have you made any efforts to prevent hallucination in long-form audio transcription (e.g., training strategies such as SFT or RL, or other approaches that are not elaborated on in the tech report)?

Thanks again for your work, and I look forward to your reply.

**Comment by gengxuelong** [comment]
Thanks for the detailed report and files—we reproduced the issue.
1. Repetition
Your script is correct. The issue comes from a training-distribution bias. Most real-world long-audio tasks require both transcription and speaker attribution, so our long-audio training data is dominated by diarized transcription. Plain long-form ASR is therefore more likely to drift off-distribution and enter decoding loops.
This prompt resolved the issue on your sample:
“听这段音频, 按时间顺序输出每一句说话人对话, 每行格式为:
`[mm:ss.d-mm:ss.d] speaker_id 转录`
非语音事件不要输出。按 start 升序, 不要额外解释。”
If you only need the transcript, remove the timestamps and speaker IDs afterward.
2. Encoder and chunking
We chose Whisper-large-v3 for its broad coverage of languages and acoustic conditions, which suits a general-purpose audio LLM.
Thirty-second chunking is not strictly required. Smaller chunks may limit local acoustic context, while larger chunks increase GPU memory usage. The key is to keep the chunking setup consistent between training and inference. We chose 30 seconds simply because most short audio clips fit within a single chunk.
3. Hallucination suppression
We mainly rely on large-scale SFT with carefully prepared long-audio data. We use no RL or techniques beyond those described in the report. In practice, data quality and task distribution matter most.

**Comment by jackyguo624** [comment]
@gengxuelong 

Thanks for your reply — the "magic prompt" worked like magic!

By the way, was this prompt used during SFT, or was it tuned specifically to fix the issue on my sample audios? (In fact, it also fixed the repetition on the other audios.)

On our in-house test set, FireRedAudio achieved impressive performance. Your answers to questions 2 and 3 were also very informative. Cheers!

**Comment by gengxuelong** [comment]
@jackyguo624 
Thanks for the kind words — really glad the prompt helped resolve the repetition issue on your other audios as well!

As mentioned earlier, our training corpus includes a large amount of long-audio speaker diarization data as a foundational task (rather than plain ASR data). To strengthen instruction-following capability, we used a diverse set of prompts for this task during training, and the one we shared is just one of many variants. We'd expect any other sentence that unambiguously expresses the same speaker-diarization intent to work similarly well. On top of that, we also support controlling the output format — you're not limited to the default one, so feel free to phrase the prompt to fit your downstream needs.

**Comment by jackyguo624** [comment]
@gengxuelong OK, catched.
FYI, I noticed a phenomenon: when the input audio is short (around 30s), the output contains no speaker labels or timestamps, but when the input audio is long, the output does include speaker labels and timestamps. I guess the long input audio triggers a switch in task type from plain ASR to diarized transcription.

**Comment by gengxuelong** [comment]
@jackyguo624  
  Do you mean the pure ASR task? We've observed the same phenomenon there, and it's expected behavior. Our ASR task is primarily focused on short-audio scenarios;  for longer audios, the underlying task shifts from ASR to speaker diarization, which is likely what's causing what you're seeing.

**Comment by jackyguo624** [comment]
@gengxuelong Yes, exactly pure ASR task.  That makes sense.
